vraptor-core/src/main/java/br/com/caelum/vraptor/converter/PrimitiveByteArrayConverter.java

```diff
@@ -0,0 +1,73 @@
+/***
+ * Copyright (c) 2009 Caelum - www.caelum.com.br/opensource
+ * All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * 	http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package br.com.caelum.vraptor.converter;
+
+import static com.google.common.base.Strings.isNullOrEmpty;
+
+import java.nio.charset.Charset;
+
+import javax.annotation.Priority;
+import javax.enterprise.context.ApplicationScoped;
+import javax.enterprise.inject.Alternative;
+import javax.inject.Inject;
+import javax.interceptor.Interceptor;
+
+import br.com.caelum.vraptor.Convert;
+import br.com.caelum.vraptor.http.EncodingHandler;
+
+/**
+ * A Byte array converter. Null or empty values are returned as an empty byte array.
+ *
+ * @author Otávio Scherer Garcia
+ * @since 4.2.0
+ */
+@Convert(byte[].class)
```

Comment:
@rponte There are an example in the pull request description. I can't provide more information about the application due a NDA. 

But I have some applications using a hard cryptography. Since Java Strings are immutable and shared by a pool, we need to use password as String so less time when possible. Using byte array or char array as good pratice when we are using a hight security, because we can override the values when we want, and Strings not.

Without this converter, in any method I need to use byte array I need to inject `EncodingHandler` to do something like this:

```
public void foo(String passwd) {
   Charset charset = Charset.fromName(encondingHandler.getCharset());
   byte[] passwdArray = passwd.getBytes(charset);
}
```

So this converter can save too much work.

At this time we have `ByteConverter`, `CharacterConverter` converters, and also for primitive types: `PrimitiveByteConverter`, `PrimitiveCharacterConverter`. Why these converters are useful and byte array isn't?